scripts/travisci/check_precommit.sh

```diff
@@ -20,31 +23,8 @@ if [[ "${TRAVIS_PULL_REQUEST}" != "false" && "${TRAVIS}" == "true" ]]; then
       cat "${commit_msg}"
     fi
 
-  done < <(git rev-list "${TRAVIS_COMMIT_RANGE}")
-else
-  git remote set-branches --add origin master
-  git fetch
-  merge_base=$(git merge-base origin/master ${TRAVIS_BRANCH})
-  pre-commit run --source ${merge_base} --origin ${TRAVIS_BRANCH}
-  status="$((${status} | ${?}))"
-
-  if [[ "${TRAVIS}" != "true" ]]; then
-  # Check commit messages
-    while read commit; do
-      commit_msg="$(mktemp)"
-      git log --format=%B -n 1 "${commit}" > "${commit_msg}"
-      pre-commit run --hook-stage commit-msg --commit-msg-file="${commit_msg}"
-      pass=$?
-      status="$((${status} | ${pass}))"
-
-      # Print message if it fails
-      if [[ "${pass}" -ne 0 ]]; then
-        echo "Failing commit message:"
-        cat "${commit_msg}"
-      fi
-
-    done < <(git rev-list ^origin/master "${TRAVIS_BRANCH}")
-  fi
+  done < <(git log --cherry-pick --left-only --pretty="%H" \
```

Comment:
For future reference, I *think* this is the same as `git log --pretty="%H" "${ORIGIN}" --not "${SOURCE}"`